terraform/modules/github-ci-bootstrap/main.tf

```diff
@@ -23,6 +23,16 @@ locals {
     cloudbuild     = "roles/cloudbuild.builds.builder"
   }
 
+  # Read-only roles for any branch
+  read_only_roles = {
+    cloudfunctions = "roles/cloudfunctions.viewer"
+    storage        = "roles/storage.objectViewer"
+    pubsub         = "roles/pubsub.viewer"
+    scheduler      = "roles/cloudscheduler.viewer"
+    run            = "roles/run.viewer"
+    cloudbuild     = "roles/cloudbuild.builds.builder" # Read-only branches still need build access
```

Comment:
You say the read-only perms are safe because they can't affect production at all, but they can still build things -- is there a risk they build things that a separate rw step then makes use of?  Is that ok?